connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1020,7 +1107,9 @@ public HerderRequest(long at, Callable<Void> action, Callback<Void> callback) {
 
         @Override
         public int compareTo(HerderRequest o) {
-            return Long.compare(at, o.at);
+            final int soonest = Long.compare(at, o.at);
+            // If tied, returning a positive value should respect insertion order.
+            return soonest != 0 ? soonest : 1;
```

Comment:
Exactly, I rejected the option of using `System.identityHashCode` because it can't guarantee uniqueness between objects (see interesting read: http://bugs.java.com/bugdatabase/view_bug.do?bug_id=6321873) 

Besides wanting to be not only practically correct but also "paranoidly" correct (again it's not hard to have two object having the same identity hash code, especially if implementations translate memory address or even when you have lots of objects in a long running JVM - hashCodes are 32-bit ints), the code as written above is ... simpler :)